x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJob.java

```diff
@@ -0,0 +1,498 @@
+ *     <li>The hostname of the worker to run the sync job.</li>
+ * </ul>
+ */
+public class ConnectorSyncJob implements Writeable, ToXContentObject {
+
+    public static final ParseField CANCELATION_REQUESTED_AT_FIELD = new ParseField("cancelation_requested_at");
+
+    public static final ParseField CANCELED_AT_FIELD = new ParseField("canceled_at");
+
+    public static final ParseField COMPLETED_AT_FIELD = new ParseField("completed_at");
+
+    public static final ParseField CONNECTOR_FIELD = new ParseField("connector");
+
+    public static final ParseField CREATED_AT_FIELD = new ParseField("created_at");
+
+    public static final ParseField DELETED_DOCUMENT_COUNT = new ParseField("deleted_document_count");
+
+    public static final ParseField ERROR_FIELD = new ParseField("error");
+
+    public static final ParseField ID_FIELD = new ParseField("id");
+
+    public static final ParseField INDEXED_DOCUMENT_COUNT_FIELD = new ParseField("indexed_document_count");
+
+    public static final ParseField INDEXED_DOCUMENT_VOLUME_FIELD = new ParseField("indexed_document_volume");
+
+    public static final ParseField JOB_TYPE_FIELD = new ParseField("job_type");
+
+    public static final ParseField LAST_SEEN_FIELD = new ParseField("last_seen");
+
+    public static final ParseField METADATA_FIELD = new ParseField("metadata");
+
+    public static final ParseField STARTED_AT_FIELD = new ParseField("started_at");
+
+    public static final ParseField STATUS_FIELD = new ParseField("status");
+
+    public static final ParseField TOTAL_DOCUMENT_COUNT_FIELD = new ParseField("total_document_count");
+
+    public static final ParseField TRIGGER_METHOD_FIELD = new ParseField("trigger_method");
+
+    public static final ParseField WORKER_HOSTNAME_FIELD = new ParseField("worker_hostname");
+
+    public static final ConnectorSyncStatus DEFAULT_INITIAL_STATUS = ConnectorSyncStatus.PENDING;
+
+    public static final ConnectorSyncJobType DEFAULT_JOB_TYPE = ConnectorSyncJobType.FULL;
+    public static final String DEFAULT_JOB_TYPE_USED_WARNING = usingDefaultValueForMissingFieldWarning(
+        JOB_TYPE_FIELD,
+        DEFAULT_JOB_TYPE.toString()
+    );
+
+    public static final ConnectorSyncJobTriggerMethod DEFAULT_TRIGGER_METHOD = ConnectorSyncJobTriggerMethod.ON_DEMAND;
+    public static final String DEFAULT_TRIGGER_METHOD_USED_WARNING = usingDefaultValueForMissingFieldWarning(
+        TRIGGER_METHOD_FIELD,
+        DEFAULT_TRIGGER_METHOD.toString()
+    );
+
+    @Nullable
+    private final Instant cancelationRequestedAt;
+
+    @Nullable
+    private final Instant canceledAt;
+
+    @Nullable
+    private final Instant completedAt;
+
+    private final Connector connector;
+
+    private final Instant createdAt;
+
+    private final Long deletedDocumentCount;
+
+    @Nullable
+    private final String error;
+
+    private final String id;
+
+    private final Long indexedDocumentCount;
```

Comment:
Any reason these can't be primitives that start at 0?

Reply:
Nope, not really. Adjusted it with https://github.com/elastic/elasticsearch/pull/102714/commits/56e5ce8bbdb1621fd4b0fa90195e2fa823b1acee. Thanks for the hint!